### PR TITLE
Use pandas Series for partition_list

### PR DESCRIPTION
**Pull request by mrocklin**
Among other things this gives us magic string support for datetime indices. 

     df = c['2010-05': '2013-02']

### CODE REVIEW

**Comment by mrocklin** [comment]
Merging in 12 hours if no comment.

**Comment by esc** [comment]
Sure, just merge it.

**Comment by esc** [comment]
@mrocklin remember to delete your feature branches when you are done. :)

File: castra/core.py [hunk 1]
```diff
@@ -190,15 +192,31 @@ def unpack_file(fn):
...
     assert key.step is None
     start, stop = key.start, key.stop
-    i = bisect(partition_list, (start, None)) if start is not None else None
-    j = bisect(partition_list, (stop, None)) + 1 if stop is not None else None
-    return i, j
+    names = list(partitions.loc[start:stop])
+
+    last = partitions.searchsorted(names[-1])[0]
+
+    stop2 = coerce_index(partitions.index.dtype, stop)
```
Comment (esc) [inline castra/core.py]: What do these lines do?

**Comment by esc** [comment]
I made some cosmetic fixes and pushed them to master.

File: castra/core.py (hunk 1 above)
Comment (mrocklin) [inline castra/core.py]: This uses loc to get out the filenames of most of the partitions.  It then checks the last value to see if we need to add one more.

**Comment by mrocklin** [comment]
> @mrocklin remember to delete your feature branches when you are done. :)

I host these branches from my own fork, not from the Blosc org

**Comment by esc** [comment]
Correct. IMHO it is considered good practice though to delete feature branches when they are merged, regardless of where they are. But of course, you are free to do how you please in your own fork.

**Pull request merged**